pollitos-league-manager

chioooo opened the pull request:
## _Pollitos-league-manager_ ##
I created 3 classes in the `model` package: `Players`, `Teams`, and `Matches`, each with their respective attributes.

I created an interface `ILeagues` to serve as the base for other league types.

I implemented two service beans using `ILeagues`, where I included the methods to be used in the endpoints.

I created two controllers:
1. **LeaguesController** -This controller contains all the endpounts.

2. **PopulatedLeaguesController** – This controller contains a PostContruct. I took the suggestions into account and modified the PopulatedLeagueController.

In `application.properties`, you can comment the profile populated to observe the difference. [src/main/resources/application.properties
](url)
To test the endpoints in Postman I put the file in the test/resources directory.
I didn't have time to implement the DTOs.

Commit:
Cambio

Commit:
Se creo una interfaz para las ligas
Se crearon los dos beans partiendo de esa interfaz
Se crearon las clases de players, matches y teams
Se creo una clase controller

Commit:
Se creo una interfaz para las ligas
Se crearon los dos beans partiendo de esa interfaz
Se crearon las clases de players, matches y teams
Se creo una clase controller

Commit:
Se creo una interfaz para las ligas
Se crearon los dos beans partiendo de esa interfaz
Se crearon las clases de players, matches y teams
Se creo una clase controller

Commit:
Se creo una interfaz para las ligas
Se crearon los dos beans partiendo de esa interfaz
Se crearon las clases de players, matches y teams
Se creo una clase controller

Adreamsa reviewed (changes requested):
You are not using DTO.

There is no property to limit the number of teams to 10

In postman, "getPlayersTeam" is wrong, it should be localhost:8080/leaguesP/soccer/player/2

I would recommend having only one endpoint folder, I see it as unnecessary to have double for the Default and Populated profiles, you can use the same ones for both.

Unlike postConstruct, LeaguesController and populatedLeaguesController are identical, so it’s all duplicate code.

I recommend re-evaluating the use of log.info and changing some of them for their variants debug and warn.

Commit:
Se creo una interfaz para las ligas
Se crearon los dos beans partiendo de esa interfaz
Se crearon las clases de players, matches y teams
Se creo una clase controller

Adreamsa commented on src/main/java/com/iwaconsolti/league/manager/controller/LeaguesController.java:
```diff
@@ -0,0 +1,129 @@
+
+    @PostMapping("/player")
+    public ResponseEntity<Players> createPlayer(@PathVariable String leagueType, @RequestBody Players newPlayer) {
+        if (leagueType.equals("soccer")) {
```
When placing anything other than soccer in the URL where the {leagueType} goes it returns baseball values, I would recommend changing the check for both specific cases and returning a 404 in case the entered value does not exist.

```suggestion
               if (leagueType.equals("soccer")) {
            Teams teamSoccer = soccerLeagues.saveTeams(newTeam);
            log.info("Team created");
            return ResponseEntity.ok(teamSoccer);
        } else if (leagueType.equals("baseball")){
            Teams teamBase = baseBallLeague.saveTeams(newTeam);
            return ResponseEntity.ok(teamBase);
        } else {
            return ResponseEntity.notFound().build();
        }
```

Commit:
Se creo una interfaz para las ligas
Se crearon los dos beans partiendo de esa interfaz
Se crearon las clases de players, matches y teams
Se creo una clase controller

Martin-Mtz commented on src/main/java/com/iwaconsolti/league/manager/service/impl/BaseBallLeagueImpl.java:
```diff
@@ -45,7 +51,7 @@ public Players findPlayers(int id) {
             if (player.getId() == id) {
                 return player;
             } else {
```
Search logic in findPlayers, findTeams and findMatches methods:

In these methods, a "Player not found", "Team not found" or "Match not found" warning message is being thrown inside the for loop, which causes a warning to be logged whenever a partial search fails. This could fill the logs with unnecessary warnings.
Solution: Move the warning out of the for loop so that it is only logged if the item is not found after the full search.

Martin-Mtz commented on src/main/java/com/iwaconsolti/league/manager/controller/LeaguesController.java:
```diff
@@ -6,13 +6,12 @@
 
+
 import java.util.List;
 
```
In the LeaguesController constructor, the parameter names soccerLeague and baseBallLeague do not match the attribute names soccerLeagues and baseBallLeague. While this works, it is recommended that variable names be consistent for readability.

Adreamsa commented on src/main/java/com/iwaconsolti/league/manager/controller/LeaguesController.java:
```diff
@@ -0,0 +1,151 @@
+    @PostMapping("/team")
+    public ResponseEntity<Teams> createTeam(@PathVariable String leagueType, @RequestBody Teams newTeam) {
+        if (leagueType.equals("soccer")) {
+            return ResponseEntity.ok(soccerLeagues.saveTeams(newTeam));
```
In the POST methods, the ResponseEntity could be changed to a 201 message to indicate that the resource was created.

```suggestion
           if (leagueType.equals("soccer")) {
            return ResponseEntity.status(HttpStatus.CREATED).body(soccerLeagues.saveTeams(newTeam));
        } else if (leagueType.equals("baseball")) {
            return ResponseEntity.status(HttpStatus.CREATED).body(baseBallLeague.saveTeams(newTeam));
        } else {
            log.warn("League not found");
            return ResponseEntity.notFound().build();
        }
```

In addition, it could be considered sending an error message when there are already 10 computers, it could be a 413 Payload Too Large or 409 Conflict message.

Adreamsa reviewed (approved):
Considers that BaseBallLeagueImpl and SoccerLeagueImpl have a lot of duplicate code

fjpgtt commented on src/main/java/com/iwaconsolti/league/manager/LsAppLeague.java:
```diff
@@ -0,0 +1,12 @@
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class LsAppLeague {
```
Why the Ls suffix?

fjpgtt commented on src/main/java/com/iwaconsolti/league/manager/service/impl/SoccerLeagueImpl.java:
```diff
@@ -0,0 +1,180 @@
+@Slf4j
+public class SoccerLeagueImpl implements ILeagues {
+
+    @Value("${league.teams.limit}")
```
i will suggest to always add a default value to avoid any problem if the settings is missing int he properties

fjpgtt commented on src/main/java/com/iwaconsolti/league/manager/service/impl/SoccerLeagueImpl.java:
```diff
@@ -0,0 +1,180 @@
+    public Players savePlayers(Players player) {
+        log.debug("Saving player {}", player);
+        player.setId(nextPlayer++);
+        players.add(player);
```
Before adding the player you should cehck if the team exists otherwise the player is created without a team

fjpgtt commented on src/main/java/com/iwaconsolti/league/manager/service/impl/SoccerLeagueImpl.java:
```diff
@@ -0,0 +1,180 @@
+        return match;
+    }
+
+    public Matches findMatches(int id) {
```
I think this method is not used

fjpgtt commented on src/main/java/com/iwaconsolti/league/manager/service/impl/SoccerLeagueImpl.java:
```diff
@@ -0,0 +1,180 @@
+
+    @Override
+    public Players savePlayers(Players player) {
+        log.debug("Saving player {}", player);
```
You should do a log.info after the entry is stored

fjpgtt commented on src/main/java/com/iwaconsolti/league/manager/service/impl/SoccerLeagueImpl.java:
```diff
@@ -0,0 +1,180 @@
+            log.warn("Team limit exceeded");
+            return null;
+        }
+        log.debug("Saving team {}", team);
```
Same as the other when an entry is created/updated/deleted is a good practice to log as info

fjpgtt commented on src/main/java/com/iwaconsolti/league/manager/service/impl/SoccerLeagueImpl.java:
```diff
@@ -0,0 +1,180 @@
+
+    @Override
+    public List<Players> getPlayersTeam(int teamId) {
+        return players.stream().filter(player -> player.getTeamId() == teamId).collect(Collectors.toList());
```
.collect(Collectors.toList) > toList()

fjpgtt commented on src/main/java/com/iwaconsolti/league/manager/service/impl/SoccerLeagueImpl.java:
```diff
@@ -0,0 +1,180 @@
+    public Teams deletePlayersTeam(int teamId) {
+        for (Teams team : teams) {
+            if (team.getId() == teamId) {
+                team.getPlayers().clear();
```
Before clearing this you will also to delete the value of each Player so they dont have a team

fjpgtt commented on src/main/java/com/iwaconsolti/league/manager/controller/PopulatedLeaguesController.java:
```diff
@@ -0,0 +1,86 @@
+import org.springframework.web.bind.annotation.*;
+
+@Profile("populated")
+@RestController
```
This should not be a RestController, it should only be a simple Component type. Since it doesnt have any endpoint declared only the posconstruc

fjpgtt reviewed (commented):
As was mention in other comments, there is a lot of duplicate code and the usage of DTOs

There are some validations that are required like the teams sent in the match request exists before the saving of the entry is done

fjpgtt commented on src/main/java/com/iwaconsolti/league/manager/controller/LeaguesController.java:
```diff
@@ -0,0 +1,151 @@
+        }
+    }
+
+    @GetMapping("/player/{teamId}")
```
Why you sometimes used the teamId and other the teamName
